tests/checks/sphinx-shared.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+set -e
```

Comment:
LGTM.
A helper file in tests/checks (or tests/pexpects/) is certainly a novelty.
The traditional approach would be to add a fish function to tests/test_functions/ but of course that's not as nice due to the lack of `set -e` and possibly others.
Eventually we might want to come up with a consistent structure for the support
files in tests/, that directory seems a bit cluttered.